### ISSUE TITLE
Custom provider gone

### ISSUE THREAD

**Issue by simonmcl** [ibm-cloud-security/appid-serversdk-nodejs#24]
I'm just starting to port my application to using Bluemix App ID instead of Mobile Client Access. Seems like the ability to add a custom provider is now gone, is this correct?

Facebook and Google authentication really doesn't cover all uses cases. Currently I have 3 apps that were only using the custom providers, is there an alternative or do I now have to build my own?

Also for my Angular 2 / webpack frontend I was using [this mobile client access](https://www.npmjs.com/package/bms-mca-oauth-sdk) module to generate oauth tokens, as I can't use the WebAppStrategy because I don't have a Node.js server, serving static files, just a html server serving 1 SPA (single page application). Will this module work with Bluemix App ID and will it continue to be supported? or will this be renamed as well?

**Comment by vitalymibm** [comment]
Yes, currently App ID does not support custom identity providers. Can you please provide us with more info about your custom idp use cases?

The old bms-mca-oauth-sdk will not work with App ID, because the token format has changed.

**Comment by simonmcl** [comment]
I'm very surprised to hear this. I also work for IBM, building applications on bluemix, while I got the notification to say it was retiring, I was not told all of this functionality would disappear, or sent a survey or asked about functionality etc. I've recently spent 3 weeks building 9 internal node modules, all of which depend on each other, several of which depend on being able to add custom identity providers ... all now useless for our next project as I can't create instances of mobile client access.

Currently i'm building applications where the end users will be using enterprise / work email accounts to log in, only certain domains will be allowed to register. It is extremely unlikely that these email addresses will be linked with facebook or google, due to the nature of the work. I would think its quite common to have similar use cases. 90% of the work i've done in the past 3 years has had similar requirements.

Will `bms-mca-oauth-sdk` be updated? or a new module created? With the lack of a custom identity provider this is the only way I could even use the service, and salvage all my work to date.
